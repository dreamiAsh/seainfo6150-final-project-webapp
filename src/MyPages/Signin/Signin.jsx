import React, {useEffect, useState} from "react";
import styles from "./Signin.module.css";
import { Link, BrowserRouter } from "react-router-dom";

const Signin = () => {

  const [submittedForm, setSubmittedForm, submitFail] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };

  return (
    <div>
      {
        submittedForm ? (
          submittedForm.get("username").length < 6 || submittedForm.get("password").length < 6? (
            <form onSubmit={onSubmit}>
              <p className={styles.alert}>Username and password should have at least 6 characters!</p>
              <div className={styles.usernameBox}>
                <label htmlFor="username">Username: </label>
                <input type="text" name="username" id="username"/>
              </div>
              <div className={styles.passwordBox}>
                <label htmlFor="password">Password: </label>
                <input type="text" name="password" id="password"/>
              </div>
              <BrowserRouter forceRefresh={true}><Link to="/signup" className={styles.signupBotton}>Sign up here</Link></BrowserRouter>
              <input className={styles.submit} type="submit" value="Sign in" />
            </form>
          ) : 
            (<p className={styles.welcome}>Welcome back {submittedForm.get("username")}!</p>)
        ) : (
          <form onSubmit={onSubmit}>
            <div className={styles.usernameBox}>
              <label htmlFor="username">Username: </label>
              <input type="text" name="username" id="username"/>
            </div>
            <div className={styles.passwordBox}>
              <label htmlFor="password">Password: </label>
              <input type="text" name="password" id="password"/>
            </div>
            <BrowserRouter forceRefresh={true}><Link to="/signup" className={styles.signupBotton}>Sign up here</Link></BrowserRouter>
            <input className={styles.submit} type="submit" value="Sign in" />
          </form>
        )
      }
    </div>
  )
}

export default Signin
