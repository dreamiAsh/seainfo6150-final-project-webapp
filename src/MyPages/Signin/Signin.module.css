.usernameBox{
    margin: 100px 37% 30px;
}
.passwordBox{
    margin: 0 37% 50px;
}
.signupBotton{
    margin: 0 37% 17px;
    color: gray;
    font-size: 14px;
    text-decoration: underline;
    display: block;
}
.submit{
    margin: 0 37% 300px;
    width: 80px;
    height: 40px;
    background-color: orange;
    border: gray;
}
.submit:hover{
    background: #E5B687;
}
.alert{
    font-weight: bold;
    margin: 100px 37%;
    color: orange
}
.welcome{
    margin: 15% 25%;
    font-size: 22px;
    font-weight: bold;
    color: grey;
}