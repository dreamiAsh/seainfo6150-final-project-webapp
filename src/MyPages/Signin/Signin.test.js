import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Signin from "./Signin.jsx";

it("renders the Signin component correctly", () => {
  const { container } = render(<Signin />);
  expect(container).toMatchSnapshot();
});