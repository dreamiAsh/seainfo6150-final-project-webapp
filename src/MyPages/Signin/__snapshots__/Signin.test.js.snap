// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Signin component correctly 1`] = `
<div>
  <div>
    <form>
      <div
        class="usernameBox"
      >
        <label
          for="username"
        >
          Username: 
        </label>
        <input
          id="username"
          name="username"
          type="text"
        />
      </div>
      <div
        class="passwordBox"
      >
        <label
          for="password"
        >
          Password: 
        </label>
        <input
          id="password"
          name="password"
          type="text"
        />
      </div>
      <a
        class="signupBotton"
        href="/signup"
      >
        Sign up here
      </a>
      <input
        class="submit"
        type="submit"
        value="Sign in"
      />
    </form>
  </div>
</div>
`;
